cmd/build/v2/tagger.go

```diff
@@ -46,110 +44,122 @@ func getTargetRegistries() []string {
 	return registries
 }
 
-// newImageTagger returns a new image tagger
+// newImageTagger returns an instance of imageTagger with the given config
 func newImageTagger(cfg oktetoBuilderConfigInterface) imageTagger {
 	return imageTagger{
 		cfg: cfg,
 	}
 }
 
-// tag returns the full image tag for the build
-func (i imageTagger) tag(manifestName, svcName string, b *model.BuildInfo) string {
+/*
+getServiceImageReference returns the image reference [name]:[tag] for the given service.
+
+When service image is set on manifest, this is one returned.
+
+Inferred tag is constructed using the following:
+[name] is the combination of the tarjetRegistry, manifestName and serviceName
+[tag] its either the buildHash or the default okteto tag "okteto"
+*/
+func (i imageTagger) getServiceImageReference(manifestName, svcName string, b *model.BuildInfo, buildHash string) string {
+	if b.Image != "" {
+		return b.Image
+	}
+
+	// build the image reference based on context and buildInfo
 	targetRegistry := constants.DevRegistry
-	sha := ""
+	tag := ""
 	if i.cfg.HasGlobalAccess() && i.cfg.IsCleanProject() {
 		targetRegistry = constants.GlobalRegistry
-		sha = i.cfg.GetBuildHash(b)
+		tag = buildHash
 	}
 	sanitizedName := format.ResourceK8sMetaString(manifestName)
-	if shouldBuildFromDockerfile(b) && b.Image == "" {
-		if sha != "" {
-			return getImageFromTmpl(targetRegistry, sanitizedName, svcName, sha)
-		}
-		return getImageFromTmpl(targetRegistry, sanitizedName, svcName, model.OktetoDefaultImageTag)
+	if serviceHasDockerfile(b) && tag != "" {
+		return useReferenceTmplate(targetRegistry, sanitizedName, svcName, tag)
 	}
-	return b.Image
+	return useReferenceTmplate(targetRegistry, sanitizedName, svcName, model.OktetoDefaultImageTag)
```

Comment:
I think we need to check in this call to `useReferenceTmplate` that `serviceHasDockerfile(b) == true` to follow previous approach

Reply:
yes! i have fix this refactor as i was returning the b.Image just when this was not empty, but also, this is the default value if there is no dockerfile in the service. I've uploaded changes.

the change here is that b.Image was returned at the end if no infer was needed, but i changed the order and the output is the same